core/icons/comment_icon.ts

```diff
@@ -155,6 +155,19 @@ export class CommentIcon extends Icon implements IHasBubble, ISerializable {
 
   /** Sets the text of this comment. Updates any bubbles if they are visible. */
   setText(text: string) {
+    let oldText: null | string = null;
+    if (this.bubbleIsVisible()) {
+      oldText = this.text;
+    }
```

Comment:
I'm confused about why this is only set when the bubble is visible. Shouldn't the old text always be the current text?

Reply:
This is here to address the fact that the [tests](https://github.com/google/blockly/blob/develop/tests/mocha/block_test.js#L1277C11-L1277C11) are expecting the default value of `oldText` to be `null`, not an empty string. 

I believe this is due to the fact that they were written expecting that the Change Event would come from `setCommentText`, which [defaults to null](https://github.com/treypisano/blockly/blob/f8a134caef1aa57cbccc429efff8ec5a561c1b77/core/block.ts#L2206) if the comment doesn't exist yet. But now that the event is coming from `setCommentText`, the default `oldText` is `this.text`, which is `''`.

It might make sense to have the tests to expect `''` as a default `oldValue`, rather than `null`.

Reply:
Mmm darn ok. We can't change the `oldValue` to be `''` because external devs could be checking for an `oldValue` of `null` to tell if the comment was just added.

In that case, I don't think this code quite works. If you open the playground, enable event logging, add a block to the workspace, and then run the following two commands in the browser console:
```
Blockly.getMainWorkspace().getTopBlocks()[0].setCommentText('first text');
Blockly.getMainWorkspace().getTopBlocks()[0].setCommentText('second text');
```
You'll see that both events have an `oldValue` of `null`, when the second event should have an `oldValue` of `'first text'`.

So I think you need to fire the event for creating a comment from `setCommentText`. This probably means you'll need to call `Blockly.Events.disable` and `Blockly.Events.enable` around the call to `comment.setText`. But I'm not sure so you'll have to check!

Sorry for all the back and forth on this issue :/ I didn't realize it was going to be so complex when I filed it! but we want to make sure the behavior is correct for developers using the library =)